pandas/core/internals/managers.py

```diff
@@ -403,6 +404,43 @@ def reduce(
             new_mgr = type(self).from_blocks(res_blocks, [self.items, index])
         return new_mgr, indexer
 
+    def grouped_reduce(
+        self: T, func: Callable, ignore_failures: bool = False
+    ) -> Tuple[T, np.ndarray]:
+        """
+        Apply grouped reduction function blockwise, returning a new BlockManager.
+
+        Parameters
+        ----------
+        func : grouped reduction function
+        ignore_failures : bool, default False
+            Whether to drop blocks where func raises TypeError.
+
+        Returns
+        -------
+        BlockManager
+        """
+        result_blocks: List[Block] = []
+
+        for blk in self.blocks:
+            try:
+                applied = blk.apply(func)
+            except (TypeError, NotImplementedError):
+                if not ignore_failures:
+                    raise
+                continue
+            result_blocks = extend_blocks(applied, result_blocks)
+
+        if len(result_blocks) == 0:
+            raise DataError("No numeric types to aggregate")
```

Comment:
i think in principle this belongs in the caller

Reply:
Yes, only with the current logic of creating the index (or inferring the length of it) in here, it needs to be done here. If the length or result index would be passed, then it can be done on the caller side

Reply:
I could also already move it upwards, and in case of empty result_blocks just use a placeholder Index